src/iOS/Xamarin.iOS/Samples/Tutorial/AuthorEditSaveMap/AuthorEditSaveMap.cs

```diff
@@ -32,102 +32,105 @@ namespace ArcGISRuntime.Samples.AuthorEditSaveMap
+            // Call the function that creates the UI.
             CreateLayout();
 
-            // Set up the AuthenticationManager
+            // Set up the AuthenticationManager.
             UpdateAuthenticationManager();
 
-            // Use the map from the view-model
+            // Use the map from the view-model.
             _mapViewModel.ResetMap();
         }
 
         public override void ViewDidLayoutSubviews()
         {
-            // Called when the layout of the view changes (e.g. phone is rotated)
-            // Define the visual frame for the MapView & Segment Buttons
-            _mapView.Frame = new CoreGraphics.CGRect(0, 0, View.Bounds.Width, View.Bounds.Height);
-            _toolbar.Frame = new CoreGraphics.CGRect(0, View.Bounds.Height - 50, View.Bounds.Width, 50);
-            _segmentButton.Frame = new CoreGraphics.CGRect(10, _toolbar.Frame.Top + 10, View.Bounds.Width - 20, 30);
+            try
+            {
+                nfloat topMargin = NavigationController.NavigationBar.Frame.Height + UIApplication.SharedApplication.StatusBarFrame.Height;
+
+                // Reposition controls.
+                _mapView.Frame = new CGRect(0, 0, View.Bounds.Width, View.Bounds.Height);
+                _toolbar.Frame = new CGRect(0, View.Bounds.Height - 40, View.Bounds.Width, 40);
+                _segmentButton.Frame = new CGRect(5, _toolbar.Frame.Top + 5, View.Bounds.Width - 10, 30);
+
+                // Update insets.
+                _mapView.ViewInsets = new UIEdgeInsets(topMargin, 0, 40, 0);
+            }
+            // Needed to prevent crash when NavigationController is null. This happens sometimes when switching between samples.
+            catch (NullReferenceException)
+            {
+            }
         }
 
         private void CreateLayout()
         {
-            // Define the Segment Button contents
-            _segmentButton.BackgroundColor = UIColor.White;
-            _segmentButton.InsertSegment("Basemap", 0, false);
-            _segmentButton.InsertSegment("New", 1, false);
-            _segmentButton.InsertSegment("Save", 2, false);
-
-            // Handle the "click" for each segment (new segment is selected)
+            // Handle the "click" for each segment (new segment is selected).
             _segmentButton.ValueChanged += SegmentButtonClicked;
 
-            // Create a toolbar on the bottom of the display
-            _toolbar = new UIToolbar();
-
-            // Add the MapView and Segment Button to the page
+            // Add the controls to the view.
             View.AddSubviews(_mapView, _toolbar, _segmentButton);
         }
 
         private void MapViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
         {
-            // Update the map view with the view model's new map
+            // Update the map view with the view model's new map.
             if (e.PropertyName == "Map" && _mapView != null)
```

Comment:
Consider using `nameof(MapViewModel.Map)` instead of hardcoding the property name. It's more refactoring-friendly.